src/pyEQL/solution.py

```diff
@@ -249,6 +249,11 @@ def __init__(
         self._solutes = solutes
         if self._solutes is None:
             self._solutes = {}
+
+        # if user specifies non-default pH AND has H+ in solutes
+        if self._pH != 7 and "H+" in self._solutes:
+            raise ValueError("Cannot specify both a non-default pH and H+ at the same time. Please provide only one.")
+
```

Comment:
Please move this logic to above line 259 (inside the `for` loop).

Also, you should run the formula through `standardize_formula` so that the error will be caught whether a user enters H+, H+1, H[+1] etc., e.g.,

```
if standardize_formula(k) == 'H[+1]' and self._pH != 7:
     <error>
```